src/main/java/com/onlinebookstore/store/repository/BookRepository.java

```diff
@@ -1,7 +1,19 @@
 package com.onlinebookstore.store.repository;
 
 import com.onlinebookstore.store.model.Book;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
 public interface BookRepository extends JpaRepository<Book, Long> {
+    @Query("FROM Book b JOIN b.categories c WHERE c.id = :categoryId")
+    List<Book> findAllByCategoryId(Long categoryId);
+
+    @Query("FROM Book b LEFT JOIN FETCH b.categories")
+    List<Book> findAllWithCategories(Pageable pageable);
+
+    @Query("FROM Book b LEFT JOIN FETCH b.categories WHERE b.id = :id")
+    Optional<Book> findBookById(Long id);
```

Comment:
Can you create a book without a category? 
Should you provide the option to create an uncategorized book?